aries/aries_vcx_core/src/wallet/entry_tag.rs

```diff
@@ -77,3 +110,42 @@ impl From<EntryTags> for Vec<EntryTag> {
         value.inner
     }
 }
+
+pub fn pair_into_entry_tag(pair: (String, String)) -> EntryTag {
+    if pair.0.starts_with('~') {
+        EntryTag::Plaintext(pair.0.trim_start_matches('~').into(), pair.1)
+    } else {
+        EntryTag::Encrypted(pair.0, pair.1)
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
The (de)serialization of `EntryTag` enum is incongruent with (de)serialization of `EntryTags` because of the default derived impls for `EntryTag` and custom `EntryTag` impls, meaning e.g. that plaintext `EntryTag` (de)serialization doesn't add (strip) the `~`, but if the `EntryTag` is a member of `EntryTags`, it does. Can you please fix that and add a test?

Reply:
Fixed by no longer deriving (de)serialization traits for `EntryTag` since tags are never serialized individually.

Reply:
I don't think this is a fix at all. These derived impls are way too easy to add. A struct using custom (de)serialize impl for it's member fields instead of relying on their own (de)serialize impl (i.e. inconsistent impls inside / outside of another struct) is clearly problematic.

Reply:
I generally agree that struct using custom (de)serialize for member fields will create problems.

However in this case, `EntryTags` is not divisible into individual fields. It is a container/wrapper and individual tags do not exist in json separately, only as a collection. `{ "a": "b" }` is not a tag, it is tags with a length of 1. I cannot (de)serialize a single `EntryTag`, same as the existing code was not (de)serializing `(String, String)`. It was always `"{ ... }" -> HashMap<String, String>`.

Reply:
No longer relevant since the tag is now a tuple.